### ISSUE TITLE
Support the new `$/` self-repository `uses:` syntax

### ISSUE THREAD

**Issue by tristanbes** [rhysd/actionlint#711]
Hi, and thanks for actionlint — we run it as a required check on our shared workflows repository, and it has caught real mistakes for us.

GitHub announced a new `uses:` syntax today (2026-07-30) that actionlint doesn't recognise yet:

> [Reference same-repository actions with self-repository syntax](https://github.blog/changelog/2026-07-30-reference-same-repository-actions-with-self-repository-syntax/)
>
> A `uses:` value that starts with `$/` resolves to your workflow's own repository at the exact commit that is running, with no checkout required.

Per the announcement it works "everywhere the workspace-relative `./` syntax works, including workflow steps, composite action steps, nested composition, and reusable workflow calls". It requires runner `2.336.0` or newer and is currently github.com only.

### Current behaviour

With actionlint 1.7.12:

```yaml
# .github/workflows/example.yml
name: Example
on: workflow_dispatch
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: $/.github/actions/my-action
```

```
example.yml:8:15: specifying action "$/.github/actions/my-action" in invalid format because ref is missing. available formats are "{owner}/{repo}@{ref}" or "{owner}/{repo}/{path}@{ref}" [action]
```

### Expected behaviour

`$/<path>` accepted without a `@ref`, since the absence of a ref is precisely the point — the action resolves to the commit already running.

### Why it matters

The syntax exists to remove a real desynchronisation risk. Referencing an action of the same repository as `owner/repo/path@v1` hardcodes a version inside the workflow, so a caller pinning `workflow.yml@v1.5.0` still loads the `@v1` action, and any push before the tag moves leaves a window where a new workflow runs against an older action.

### Workaround in the meantime

For anyone landing here, `-ignore` scoped to this single message keeps the rest of the linting intact:

```bash
actionlint -ignore 'specifying action "\$/.+" in invalid format because ref is missing'
```

Happy to help test a fix if that's useful. Thanks again for the tool.

**Comment by muzimuzhi** [comment]
> For anyone landing here, `-ignore` scoped to this single message keeps the rest of the linting intact:
> 
> actionlint -ignore 'specifying action "\$/.+" in invalid format because ref is missing'

Similarly, the `paths.<path-glob>.ignore` [config setting](https://github.com/rhysd/actionlint/blob/main/docs/config.md):

```yaml
paths:
  .github/workflows/**/*.{yaml,yml}:
    ignore:
      - reusable workflow call "\$/.+" at "uses" is not following the format
      - specifying action "\$/.+" in invalid format because ref is missing
```

_Update:_ Added `ignore` config for `$/` self-reference reusable workflows.

**Comment by silverwind** [comment]
This is needed, when running `zizmor` together with this linter, the linters disagree and zizmor wants `$/` syntax while actions lint does not support it and fails with `specifying action "\$/.+" in invalid format because ref is missing`. This came in https://github.com/zizmorcore/zizmor/pull/2271.
